### PR TITLE
Small Cleanup

### PR DESCRIPTION
**Pull request by cademoor**
Handle exceptions in a general way (in Controllers). Simplify OrganizationUser representation to only have one source of truth for foreign keys #125 and #126

### CODE REVIEW

File: Src/Common/Domain/service/null/NullOrganizationUserService.cs [hunk 1]
```diff
@@ -19,24 +19,18 @@ public virtual void AddOrganizationUser(IOrganizationUser organizationUser)
...
-        public virtual IOrganizationUser GetOrganizationUser(int organizationId, int userId)
-        {
-            return null; 
-        }
-        public virtual int GetOrganizationUserRole(int organizationId, int userId)
-        {
-            return -1; 
         }
+
         public virtual IOrganizationUser[] GetOrganizationUsers()
```
Comment (ErikAggie) [inline Src/Common/Domain/service/null/NullOrganizationUserService.cs]: Wonder if we even want this. Don't change it now, but unless we want admins to see all of these types of users we probably won't need this method.

File: Src/Service/Service/service/OrganizationService.cs [hunk 2]
```diff
@@ -45,6 +45,9 @@ public virtual void RemoveOrganization(int recordId)
                 return;
             }
 
+            IOrganizationUser[] organizationUsers = UnitOfWork.OrganizationUsers.FindBy(ou => ou.Organization == organization);
+            UnitOfWork.OrganizationUsers.RemoveAll(organizationUsers);
+
```
Comment (ErikAggie) [inline Src/Service/Service/service/OrganizationService.cs]: Believe we need to do the same thing for Projects here. They're tied to organizations, and shouldn't survive without the organization that created them.

File: Src/Common/Domain/service/null/NullOrganizationUserService.cs (hunk 1 above)
Comment (jsorbo) [inline Src/Common/Domain/service/null/NullOrganizationUserService.cs]: Even if it's not a strict requirement we've defined, it seems helpful for allowing admins to assign roles to other organization users. I vote we leave it in.

Comment (cademoor) [inline Src/Common/Domain/service/null/NullOrganizationUserService.cs]: Thanks to you both.

**Pull request merged**